### PR TITLE
Remove `isIgnored` from the public API of Mutator class. Make methods final

### PR DESCRIPTION
**Pull request by borNfreee**
This PR:

- [x] Fixes https://github.com/infection/infection/issues/251
- [x] Covered by tests

The main idea is to reduce public API of `Mutator` class and improve future maintainability. Read more in #251

### CODE REVIEW

File: src/Mutator/Util/Mutator.php [hunk 1]
```diff
@@ -25,17 +26,27 @@ public function __construct(MutatorConfig $config)
...
+        return !$this->config->isIgnored($reflectionClass->getName(), $node->getAttribute(ReflectionVisitor::FUNCTION_NAME, ''));
     }
 
-    public function isIgnored(string $class, string $method): bool
+    final public static function getName(): string
     {
-        return $this->config->isIgnored($class, $method);
+        $parts = explode('\\', static::class);
+
+        return $parts[count($parts) - 1];
```
Comment (sanmai) [inline src/Mutator/Util/Mutator.php]: While we are at it, this looks very much like 
```php
return end($parts);
```
Or even
```php
return substr(static::class, strrpos(static::class, '\\') + 1);
```

**Comment by sanmai** [comment]
@BackEndTea  `shouldMutate` includes `mutatesNode` which [is the fastest method here](https://github.com/infection/infection/pull/235#discussion_r176333456); no order change needed.

But surely this can be confusing.

**Comment by BackEndTea** [comment]
In that case i have no objections

**Comment by borNfreee** [comment]
@sanmai @BackEndTea rebased and addressed comment

**Comment by BackEndTea** [comment]
👍 for me

**Comment by borNfreee** [comment]
Thank you @sanmai & @BackEndTea

**Pull request merged**